File: src/Company_System.Core/DTO/Auth/CreateAccountDTO.cs
using System.ComponentModel.DataAnnotations;
using HR_System.Core.ENUM;

namespace HR_System.Core.DTO.Auth;

public class CreateAccountDTO
{
    [Required]
    public required string Email { get; set; }
    [Required]
    public required string Password { get; set; }
    [Required]
    public required string UserName { get; set; }
    [Required]
    public required string FullName { get; set; }
    [Required]
    public required string PhoneNumber { get; set; }
    [Required]
    public required RolesEnum Role { get; set; }
    
    
    // override

    public override string ToString()
    {
        return
            $"Email: {Email}\nPassword: {Password}\nUserName: {UserName}\nFullName: {FullName}\nPhoneNumber: {PhoneNumber}\nRole: {Role.ToString()}";
    }
}

File: FrontEnd/Company_System/src/core/dto/create-account-dto.ts
import { PositionsEnum } from '../enum/positions-enum';

export interface AddEmployeeDTO {
  email: string;
  password: string;
  userName: string;
  fullName: string;
  phoneNumber: string;
  position: PositionsEnum;
  parentId: string;
  image: File;
}
